aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -2257,14 +2375,14 @@ impl AptosVM {
         )?;
 
         match payload {
-            TransactionPayload::Script(_) | TransactionPayload::EntryFunction(_) => {
-                transaction_validation::run_script_prologue(
-                    session,
-                    txn_data,
-                    log_context,
-                    traversal_context,
-                )
-            },
+            TransactionPayload::Script(_)
+            | TransactionPayload::EntryFunction(_)
+            | TransactionPayload::Automation(_) => transaction_validation::run_script_prologue(
```

Comment:
I believe we need to customize here. We have to call a different prologue, the one which does not check for `sequence_number` of `Account`.

Reply:
Automation is the payload of the transaction that aims to register the automation task, so i named it/refer to it  `AutomationTransaction`. So this type of transactions is considered normal user transaction with single exception which is that inner-payload/automation-task-payload should be validated. so senders sequence number will be increased in this case.
`AutoamtedTransaction` is the transaction that will be constructed based on the registered tasks, and for this transactions prologue and epologue will be different, and sequence number check and increment will be omitted: 
The draft PR for `AutomatedTranaction` processing flow can be found here:  https://github.com/Entropy-Foundation/aptos-core/pull/136

Reply:
@aregng , Ah I see. I wonder if in that case we should rename `AutomationTransaction` to `AutomationRegistration`. The names `AutomationTransaction` and `AutomatedTransaction` are so similar that it may be hard for a code reader to read/comprehend the code easily. What do you think?

Reply:
Sounds good.